packages/vite/src/node/utils.ts

```diff
@@ -303,6 +303,7 @@ export function getPotentialTsSrcPaths(filePath: string): string[] {
 }
 
 const importQueryRE = /(\?|&)import=?(?:&|$)/
+const directRequestRE = /(\?|&)direct(?:&|$)/
```

Comment:
We could add the same guard we have for `import` here. It looks like Vercel and maybe other services will add a `=` after a query without value.
```suggestion
const directRequestRE = /(\?|&)direct=?(?:&|$)/
```